modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMStructureLocalServiceTest.java

```diff
@@ -286,6 +311,73 @@ public void testCopyStructure() throws Exception {
 		Assert.assertEquals(structure.getType(), copyStructure.getType());
 	}
 
+	@Test
+	public void testDDMStructureWithSameExternalReferenceCodeBetweenDifferentSites()
+		throws Exception {
+
+		DDMStructure structure1 = _addStructure(RandomTestUtil.randomString());
+
+		User user = UserTestUtil.addGroupAdminUser(GroupTestUtil.addGroup());
+
+		DDMStructure structure2 = _addStructure(
+			structure1.getExternalReferenceCode(), user.getGroupId(),
+			RandomTestUtil.randomString(), user.getUserId());
+
+		Assert.assertNotNull(
+			_ddmStructureLocalService.fetchDDMStructureByExternalReferenceCode(
+				structure2.getExternalReferenceCode(),
+				structure2.getGroupId()));
+
+		Assert.assertNotEquals(structure1, structure2);
+
+		Assert.assertEquals(
+			1,
+			_ddmStructureLocalService.getStructuresCount(
+				structure1.getGroupId()));
+
+		Assert.assertEquals(
+			1,
+			_ddmStructureLocalService.getStructuresCount(
+				structure2.getGroupId()));
+
+		_ddmStructureLocalService.deleteStructureByExternalReferenceCode(
+			structure1.getExternalReferenceCode(), structure1.getGroupId());
+
+		Assert.assertEquals(
+			0,
+			_ddmStructureLocalService.getStructuresCount(
+				structure1.getGroupId()));
+
+		Assert.assertEquals(
+			1,
+			_ddmStructureLocalService.getStructuresCount(
+				structure2.getGroupId()));
+	}
+
+	@Test
+	public void testDeleteDDMStructureByExternalReferenceCode()
+		throws Exception {
+
+		DDMStructure ddmStructure = _addStructure(
+			RandomTestUtil.randomString());
+
+		_ddmStructureLocalService.deleteStructureByExternalReferenceCode(
+			ddmStructure.getExternalReferenceCode(), ddmStructure.getGroupId());
+
+		Assert.assertNull(
+			_ddmStructureLocalService.fetchDDMStructureByExternalReferenceCode(
+				ddmStructure.getExternalReferenceCode(),
+				ddmStructure.getGroupId()));
+	}
+
+	@Test(expected = NoSuchStructureException.class)
+	public void testDeleteNonexistingDDMStructureByExternalReferenceCode()
```

Comment:
You should cover this scenario inside the method `testDeleteDDMStructureByExternalReferenceCode()` using `AssertUtils.assertFailure`.